docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-    tm_impl->addEventListener(my_tm_listener)
+  Json::Value& dict = last_state().get_dictionary();
+  dict["TransportManager"]["TcpAdapter"]["devices"] = devices_dictionary;
 }
 ```
 
-## Adding Custom Transport Adapters to the Default TM
-
-!!! note
-
-About Transport Adapters in general:
-
-  * TM may contain zero to N Transport Adapters (a TM with 0 TAs does not do anything)
-  * Each adapter corresponds to a specific type of transport (Bluetooth, USB, TCP, etc.)
-  * TA implements transport specific search, connect, disconnect, and data transfer routines
-
-!!!
-
-!!! note
-
-Several instances of TA:
-
-  * Several TAs of the same type can exist in the TM
-  * The results of using several instances of the default TAs is not defined
-  * It is the developer's responsibility to create a custom Adapter that operates well under these conditions
-
-!!!
-
-!!! note
-
-Custom TA
-
-The simplest way to add a custom TA is to derive it from the existing implementation of the TCP or BT adapter and add it to the TM
-
-!!!
-
-```
-#include "transport_manager.h"
-#include "transport_manager_default.h"
-
-{
-    // The default implementation of TCP adapter is used in this example
-    // The developer is free to create their own implementation
-    TransportAdapter *my_transport_adapter = static_cast<TransportAdapter*>(new TcpTransportAdapter);
-    my_transport_adapter->init(); // note: TA must be initialized before getting an instance of TM
-    TransportManager *tm_impl = TransportManagerDefault::instance(); // note: When custom TA is added to the TM, the default listener will be assigned to it automatically
-    tm_impl->addTransportAdapter(my_transport_adapter)
-}
-
-```
+#### Transport Adapter Structure
 
-### Initialization
+|||
+Figure 3: Transport Adapter UML Diagram
+![TA](./assets/ta.png)
+|||
 
-By default, the TA is initialized during TM initialization. If the TA needs to be added to a TM which has already been initialized, the TA should be initialized before it is added the TM
+### Transport Adapter Workers
 
-!!! note
+The Device Scanner is responsible for scanning for new devices to connect with. When a device is found, this worker is responsible for alerting the Transport Adapter, as well as alerting the Transport Manager via the Transport Adapter Listener. Next, the Transport Manager will instruct the Transport Adapter to connect with the devices.
 
-The TM has a TCP adapter by default. Adding the new instance of the same Adapter may result in unexpected behavior. The developer must change the standard behavior of the TCP adapter to elinatate any potential problems.
+The Client Connection Listener implements receiving a connection that is originated by a device. This will typically wait for connection from a device, then establish that connection, finally alerting the Transport Manager via the Transport Manager Listener of the newly connected device and app IDs.
```

Comment:
Mention USB and TCP are examples of adapters that implement this scheme.

Reply:
I mention this in the subcomponent's section, should I also add it to the summary here?

Reply:
same as above, up to you.